docs/how-to-write-function.md

```diff
@@ -0,0 +1,89 @@
+
+## Write a Function
+
+Let's take a closer look at `PaymentFunction` to better understand how to write a function.
+
+```java
+public class PaymentFunction extends Function {
+
+  @Override
+  public void invoke(
+      Database database,
+      Optional<JsonObject> functionArgument,
+      JsonObject contractArgument,
+      Optional<JsonObject> contractProperties) {
+    JsonArray array = contractArgument.getJsonArray(ASSETS_ATTRIBUTE_NAME);
+    int amount = contractArgument.getInt(AMOUNT_ATTRIBUTE_NAME);
+    String fromId = array.getString(0);
+    String toId = array.getString(1);
+
+    Optional<Result> account1 = database.get(new Get(new Key(new TextValue("id", fromId))));
+    Optional<Result> account2 = database.get(new Get(new Key(new TextValue("id", toId))));
+
+    long balance1 = ((BigIntValue) account1.get().getValue("balance").get()).get();
+    long balance2 = ((BigIntValue) account2.get().getValue("balance").get()).get();
+
+    // Transfer
+    balance1 -= amount;
+    balance2 += amount;
+
+    Put put1 =
+        new Put(new Key(new TextValue("id", fromId)))
+            .withValue(new BigIntValue("balance", balance1));
+    Put put2 =
+        new Put(new Key(new TextValue("id", toId)))
+            .withValue(new BigIntValue("balance", balance2));
+
+    database.put(put1);
+    database.put(put2);
+  }
+}
+```
+
+As you can easily see, it is a money transfer application, which sends the specified amount of moneny in between two accounts.
+
+### About the arguments
+
+Similar to a Contract using `Ledger` object to manage assets, a Function uses `Database` object to manage records in the underlining database. Note that `Database` implements [Scalar DB](https://github.com/scalar-labs/scalardb) interface so that you can do CRUD operations base on [the data model](https://github.com/scalar-labs/scalardb/blob/master/docs/design.md#data-model) of Scalar DB. 
+
+`JsonObject` for a function argument is a json object and a runtime argument for the function specified by the requester. The argument is not digitally signed as opposed to the contract argument so that it can be used to pass data that is stored in the database but it might be deleted at some later point for some reason.
+
+`contractArgument` and `contractProperties` are the corresponding contract's argument and properties. See [the contract guide](how-to-write-contract.md) to understand what they are.
+
+### How Functions and Contracts are tied together
+
+Functions to be executed are specified in a contract argument with `__functions__` json key. As described in [the contract guide](how-to-write-contract.md#write-a-complex-contract), a contract can invoke another contract, so multiple contracts and multiple functions can be grouped together.
+
+Scalar DL executes the group of opeations in a ACID manner so that they can be done atomically and in a consistent, isolated and durable manner.
```

Comment:
```suggestion
Scalar DL executes the group of operations in an ACID manner so that they can be done atomically and in a consistent, isolated, and durable manner.
```